presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java

```diff
@@ -113,7 +115,11 @@ public boolean hasNext()
             {
                 boolean hasNext = iterator.hasNext();
                 if (!hasNext) {
-                    cache.put(path, ImmutableList.copyOf(files));
+                    runtimeStats.addMetricValue(FILES_READ_COUNT, NONE, files.size());
+                    if (isCached && path.isPresent()) {
```

Comment:
Checking for path is present seems to an additional logic, is there some case which led to this addition?

Reply:
Not really. Just an additional check. I can remove if it's not necessary/ could potentially cause problems.